src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/OcrValidator.java

```diff
@@ -83,14 +89,51 @@ public OcrValidator(
                         authTokenGenerator.generate()
                     ))
                     .onSuccess(res -> handleValidationResponse(res, envelope, docWithOcr))
-                    .onFailure(exc -> handleRestClientException(exc, validationUrl, envelope, docWithOcr))
+                    .onFailure(exc -> handleRestClientException(
+                        exc,
+                        validationUrl,
+                        envelope,
+                        docWithOcr
+                    ))
                     .map(response -> ocrValidationWarnings(docWithOcr, response.warnings))
-                    .getOrElseGet(exc ->
-                        ocrValidationWarnings(
-                            docWithOcr,
-                            singletonList("OCR validation was not performed due to errors")
-                        )
-                    )
+                    .getOrElseGet(exc -> handleOcrValidationError(
+                        exc,
+                        blobClient,
+                        leaseId,
+                        docWithOcr,
+                        validationUrl,
+                        envelope.zipFileName
+                    ))
+        );
+    }
+
+    private OcrValidationWarnings handleOcrValidationError(
+        Throwable ex,
+        BlobClient blobClient,
+        String leaseId,
+        InputScannableItem docWithOcr,
+        String validationUrl,
+        String zipFileName
+    ) {
+        if (ex instanceof HttpServerErrorException) {
+            if (ocrValidationRetryManager.setRetryDelayIfPossible(blobClient, leaseId)) {
```

Comment:
should both this scenario (maximum number of retries) and exceptions different than `HttpServerErrorException` be handled the same exact way?

Reply:
Yes. In the existing code any http status response code is handled by raising OCR validation warnings. After this change in case of 5xx status response code will cause several retries after which the workflow will fall back to raising OCR validation warnings.